astropy/cosmology/core.py

```diff
@@ -1895,6 +1921,175 @@ def _T_hypergeometric(self, x):
         from scipy.special import hyp2f1
         return 2 * np.sqrt(x) * hyp2f1(1./6, 1./2, 7./6, -x**3)
 
+    def _dS_age(self, z):
+        """ Age of the universe in Gyr at redshift ``z``.
+
+        The age of a de Sitter Universe is infinite.
+
+        Parameters
+        ----------
+        z : array-like
+          Input redshifts.
+
+        Returns
+        -------
+        t : `~astropy.units.Quantity`
+          The age of the universe in Gyr at each input redshift.
+
+        See Also
```

Comment:
This isn't your fault, since we (probably I) made this mistake elsewhere, but you can definitely omit some of the 'bonus' documentation for inner _ methods.  Really, you could probably kill all the formal docs on the interior methods, but we should do that in another pull.

Reply:
This seemingly minor docstring point actually motivated me to realize that I should inject a layer to be redefined for `age` and `lookback_time`.  I've done so and now `age` calls `_age`, and `_age` is what gets redefined (and same for `lookback_time`, `_lookback_time`).  This preserves the documentation of the public method while still allowing for re-definitions based on optimizations.